eng/scripts/run_tests.ps1

```diff
@@ -9,9 +9,7 @@ Push-Location sdk/$serviceDirectory
 Write-Host "##[command] Executing 'go test -timeout $testTimeout -v -coverprofile coverage.txt ./...' in sdk/$serviceDirectory"
 
 go test -timeout $testTimeout -v -coverprofile coverage.txt ./... | Tee-Object -FilePath outfile.txt
-if ($LASTEXITCODE) {
-    exit $LASTEXITCODE
-}
+# go test will return a non-zero exit code on test failures so don't skip generating the report in this case
```

Comment:
Would you still want to capture the non-zero exit code and defer the failure to the end of the script, that way the test task still fails and can be tracked.

Reply:
Any failed tests will still cause CI to fail.  E.g. we don't check the exit code for `go test` in other (legacy) repos and things work as expected.

Reply:
I agree that capturing the non-zero and returning that after we generate the report would be safer.

Reply:
It shouldn't be necessary though.  E.g. we don't do this for [`go-autorest`](https://github.com/Azure/go-autorest/blob/main/azure-pipelines.yml#L68) and CI works as expected.

Reply:
I agree it's not necessary for failing CI. Capturing failures at the specific step in which they failed is helpful for error tracking, e.g. when we need to formulate historical queries to track when/how a certain step failed. And I find it's also useful to discern quickly whether the tests themselves failed vs. some engsys component of the coverage checking failed. Basically when we lose granularity it makes differentiating engsys vs. language failures a little bit harder in some cases.